src/nplinker/pairedomics/podp_antismash_downloader.py

```diff
@@ -25,81 +31,228 @@
+
+    for i, genome_record in enumerate(genome_records):
+        # get the best available ID from the dict
+        assert isinstance(genome_record['genome_ID'], dict)
+        raw_genome_id = _get_best_available_genome_id(
+            genome_record['genome_ID'])
+        if raw_genome_id is None:
+            logger.warning(
+                f'Ignoring genome record "{genome_record}" due to missing genome ID field'
+            )
+            continue
+
+        # use this to check if the lookup has already been attempted and if
+        # so if the file is cached locally
+        if raw_genome_id not in genome_status:
+            genome_status[raw_genome_id] = GenomeStatus(raw_genome_id, "None")
+
+        genome_obj = genome_status[raw_genome_id]
+
+        logger.info(
+            f'Checking for antismash data {i + 1}/{len(genome_records)}, current genome ID={raw_genome_id}'
+        )
+        # first check if file is cached locally
+        if Path.exists(Path(genome_obj.filename)):
+            # file already downloaded
+            logger.info(
+                f'Genome ID {raw_genome_id} already downloaded to {genome_obj.filename}'
+            )
+            genome_record['resolved_refseq_id'] = genome_obj.resolved_refseq_id
+        elif genome_obj.attempted:
+            # lookup attempted previously but failed
+            logger.info(
+                f'Genome ID {raw_genome_id} skipped due to previous failure')
+            genome_record['resolved_refseq_id'] = genome_obj.resolved_refseq_id
+        else:
+            # if no existing file and no lookup attempted, can start process of
+            # trying to retrieve the data
+
+            # lookup the ID
+            logger.info(
+                f'Beginning lookup process for genome ID {raw_genome_id}')
+
+            assert isinstance(_resolve_refseq_id(genome_record['genome_ID']),
+                              str)
+            genome_obj.resolved_refseq_id = _resolve_refseq_id(
+                genome_record['genome_ID'])
+            genome_obj.attempted = True
+
+            if genome_obj.resolved_refseq_id == "":
+                # give up on this one
+                logger.warning(f'Failed lookup for genome ID {raw_genome_id}')
+                genome_obj.add_line(genome_status_file)
+                continue
+
+        # if we got a refseq ID, now try to download and extract the data from antismash
+        download_and_extract_antismash_data(genome_obj.resolved_refseq_id,
+                                            project_download_cache,
+                                            project_file_cache)
+
+        genome_obj.add_line(genome_status_file)
+
+        output_path = Path(project_file_cache, 'antismash',
+                           genome_obj.resolved_refseq_id)
+        if not Path.exists(Path(output_path, 'completed')):
+            with open(Path(output_path, 'completed'), 'w'):
+                pass
```

Comment:
You can use `Path.touch (output_path / 'completed', exist_ok = False)`

Reply:
With Path.touch if the file already exists, FileExistsError is raised. I think we don't want this, right? @CunliangGeng

Reply:
Use `exist_ok = True` should fix it.